unittests/unit/SharedLibraryWamIkFast.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2017, Graphics Lab, Georgia Tech Research Corporation
+ * Copyright (c) 2017, Personal Robotics Lab, Carnegie Mellon University
+ * All rights reserved.
+ *
+ * This file is provided under the following "BSD-style" License:
+ *   Redistribution and use in source and binary forms, with or
+ *   without modification, are permitted provided that the following
+ *   conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef DART_TEST_SHAREDLIBRARYWAMIKFAST_HPP_
+#define DART_TEST_SHAREDLIBRARYWAMIKFAST_HPP_
+
+#include "dart/common/SharedLibrary.hpp"
+#include "dart/dynamics/IkFast.hpp"
+
+class SharedLibraryWamIkFast : public dart::dynamics::IkFast
```

Comment:
Is this the code that an end-user would implement to use IKFast on their robot? Why is this necessary versus just passing the path to the `.so` file into one of the `SharedLibrary` functions?

Reply:
An end-user would juse use this class. This class enables the user to use DART's inverse kinematics framework with IkFast `.so` file without any further modification. All the user needs to do is passing the path to the `.so` file and the list of free/non-free joint indices like:
```cpp
  std::string libName = "libGeneratedWamIkFast.so";
  std::vector<std::size_t> ikFastDofs{0, 1, 3, 4, 5, 6};
  std::vector<std::size_t> ikFastFreeDofs{2};
  ik->setGradientMethod<dynamics::SharedLibraryIkFast>(
        libName, ikFastDofs, ikFastFreeDofs);
```

`SharedLibrary` only loads arbitrary `.so` file and extracts symbols from the file, but doens't have any information how to interact with IkFast nor the inverse kinematics framework.

Reply:
@jslee02 Could you give an example how to generate ```.so``` file with wam robot?

Reply:
Once you have a `.cpp` file generated for WAM, you just compile the file with any compiler you like. Hope [this](http://openrave.org/docs/0.8.2/openravepy/ikfast/#compiling-with-gcc) helps. Note https://github.com/dartsim/dart/blob/master/examples/osgExamples/osgWamIkFast/ikfast/ikfast71.Transform6D.4_6_9_10_11_12_f8.cpp is just an example of IK for WAM. You might want different [free joints](http://openrave.org/docs/0.8.2/openravepy/ikfast/#details) setting for the same robot.